nautobot/docs/apps/migrating-jobs-from-nautobot-v1.md

```diff
@@ -0,0 +1,157 @@
+
+#### Read-Only Meta Attribute
+
+The `read_only` Job field no longer changes the behavior of Nautobot core and is informational only in v2.0.
+
+### Job Registration
+
+All Jobs must be registered in the celery task registry to be available in Nautobot. This must be accomplished by calling `nautobot.core.celery.register_jobs(*job_classes)` at the top level of a Job module so that it is registered when the module is imported. The `register_jobs` method accepts one or more job classes as arguments.
+
+!!! example
+    ```py
+    from nautobot.core.celery import register_jobs
+    from nautobot.extras.jobs import Job
+
+    class MyJob(Job):
+        def run(self):
+            pass
+
+    register_jobs(MyJob)
+    ```
+
+### Job Logging
+
+All of the custom `log` methods have been removed from the Job class. Instead, use either the Python logger from `nautobot.extras.jobs.get_task_logger` or use `self.logger` in the Job class.
+
+The following table describes the mapping of the old log methods to the new methods:
+
+| Old Method         | New Method             |
+| ------------------ | ---------------------- |
+| `self.log_debug`   | `self.logger.debug`    |
+| `self.log_info`    | `self.logger.info`     |
+| `self.log_success` | `self.logger.info`     |
+| `self.log_warning` | `self.logger.warning`  |
+| `self.log_failure` | `self.logger.error`    |
+| `self.log`         | `self.logger.info`     |
+| No equivalent      | `self.logger.critical` |
+
+Some logging features from v1.X are accessible when passing a `dict` to the `extra` kwarg in any logger method:
+
+* `logger.debug("message", extra={"skip_db_logging": True})` - Log the message to the console but not to the database
+* `logger.debug("message", extra={"object": obj})` - Replaces the `obj` kwarg in Nautobot v1.X Job logging methods
+* `logger.debug("message", extra={"grouping": "string"})` - Replaces the `active_test` Job property in Nautobot v1.X
+
+For more detailed documentation on Job logging see the [Job Logging](../additional-features/jobs.md#logging) section of the Jobs feature documentation.
+
+### Tracking Job State
+
+`JobResult.status` is now automatically tracked by celery. Job authors should no longer manually change `self.job_result.status` or `self.job_result.completed` and should instead raise an exception if the Job status should be set to failed (the failed status is now `"FAILURE"`).
+
+The Job's `self.failed` flag that was used to determine if a Job failed has been removed. This flag was set to `True` when the `log_failure` method was called. Job authors should track their Job's internal state and raise an exception to fail the Job.
+
+### Request Property
+
+The `request` property has been changed to a celery request instead of a Django request and no longer includes the information from the web request that initiated the Job. The `user` object is now available as `self.user` instead of `self.request.user`.
```

Comment:
```suggestion
The `request` property has been changed to a Celery request instead of a Django request and no longer includes the information from the web request that initiated the Job. The `user` object is now available as `self.user` instead of `self.request.user`.
```